robotbase/src/main/java/frc/robot/subsystems/ShooterPivot.java

```diff
@@ -64,26 +68,74 @@ public void stop() {
 
   public void resetEncoder() {
     m_pivotMotor.getEncoder().setPosition(0);
-    m_targetRotations = 0;
+    m_targetRotation = 0;
   }
 
   public double getPivotRotations() {
     return m_pivotMotor.getEncoder().getPosition();
   }
 
-  public double getAngle() {
+  public double getPosition() {
     return m_absoluteEncoder.getPosition();
   }
 
-  public boolean isPivotAtRotations() {
+  public boolean isPivotAtRotation() {
     return Utility.isWithinTolerance(
-        getAngle(), m_targetRotations, SHOOTER_PIVOT.POSITION_ALLOWED_ERROR);
+        getPosition(), m_targetRotation, SHOOTER_PIVOT.POSITION_ALLOWED_ERROR);
+  }
+
+  private boolean hasTarget() {
+    return Robot.shooterLimelight.validTargetExists();
   }
 
   @Override
   public void periodic() {
-    if (m_isClosedLoopEnabled && isPivotAtRotations()) {
-      m_isClosedLoopEnabled = false;
+    if (m_isClosedLoopEnabled) {
+      // Calculate feedforward and set pid reference
+      // Use 2023 ArmRotation as example
+    }
```

Comment:
@NolanCampbell2357 originally, in this subsystem, we were using `ControlType.kPosition`. In the arm rotation last year, we were using smart motion. Should we try to model this subsystem off of the arm rotation from last year instead of how we were originally doing it?

Reply:
No, I believe we want kPosition to use with the helium absolute encoder